ts/editor/EditorField.svelte

```diff
@@ -0,0 +1,104 @@
+<!--
+Copyright: Ankitects Pty Ltd and contributors
+License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+-->
+<script context="module" lang="ts">
+    import type { EditingAreaAPI } from "./EditingArea.svelte";
+    import contextProperty from "../sveltelib/context-property";
+
+    export interface FieldData {
+        name: string;
+        fontFamily: string;
+        fontSize: number;
+        direction: "ltr" | "rtl";
+    }
+
+    export interface EditorFieldAPI {
+        element: HTMLElement;
+        index: number;
+        direction: "ltr" | "rtl";
+        editingArea?: EditingAreaAPI;
+    }
+
+    const key = Symbol("editorField");
+    const [set, getEditorField, hasEditorField] = contextProperty<EditorFieldAPI>(key);
```

Comment:
Not a change needed for this PR (if at all), but would it make sense to have contextProperty() take a string and create the symbol itself?